staging/src/k8s.io/kubectl/pkg/cmd/portforward/portforward.go

```diff
@@ -420,10 +420,14 @@ func (o PortForwardOptions) RunPortForwardContext(ctx context.Context) error {
 	signal.Notify(signals, os.Interrupt)
 	defer signal.Stop(signals)
 
+	returnCtx, returnCtxCancel := context.WithCancel(context.Background())
```

Comment:
if you inherit the context, then you should just need only `case <-returnCtx.Done():` in the select